### PR TITLE
tallgeese.0.1 - via opam-publish

### PR DESCRIPTION
**Pull request by fxfactorial**
Enhanced SSH on OS X

Enhanced SSH experience on OS X using native Cocoa tallgeese is an
OCaml/Objective-C application that makes the SSH experience more
enjoyable on OS X.

---
* Homepage: http://hyegar.com
* Source repo: https://github.com/fxfactorial/tallgeese.git
* Bug tracker: https://github.com/fxfactorial/tallgeese/issues

---

Pull-request generated by opam-publish v0.3.1

### CODE REVIEW

**Comment by avsm** [comment]
```
# [...]
# ocamlfind: Package `cmdliner' not found
# W: Field 'pkg_cmdliner' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d cmdliner > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-86fe34.txt'' terminated with error code 2
# ocamlfind: Package `ssh' not found
# W: Field 'pkg_ssh' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d ssh > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-29e051.txt'' terminated with error code 2
# ocamlfind: Package `maxminddb' not found
# W: Field 'pkg_maxminddb' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d maxminddb > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-e8dcbd.txt'' terminated with error code 2
# E: Cannot find findlib package cmdliner
# E: Cannot find findlib package maxminddb
# E: Cannot find findlib package ssh
# E: Failure("3 configuration errors")
```

**Comment by fxfactorial** [comment]
Okay, that's odd to me.

**Comment by avsm** [comment]
Could you merge this against the current master branch? I've merged libssh now.  It'll probably fail because of missing symbols, but that's ok for now.

**Comment by samoht** [comment]
@avsm indeed:

```
#=== ERROR while installing tallgeese.0.1 =====================================#
# opam-version 1.2.2
# os           darwin
# command      ocaml setup.ml -configure --prefix /Users/travis/.opam/system
# path         /Users/travis/.opam/system/build/tallgeese.0.1
# compiler     system (4.02.3)
# exit-code    1
# env-file     /Users/travis/.opam/system/build/tallgeese.0.1/tallgeese-2625-8b8e2c.env
# stdout-file  /Users/travis/.opam/system/build/tallgeese.0.1/tallgeese-2625-8b8e2c.out
# stderr-file  /Users/travis/.opam/system/build/tallgeese.0.1/tallgeese-2625-8b8e2c.err
### stderr ###
# [...]
# ocamlfind: Package `cmdliner' not found
# W: Field 'pkg_cmdliner' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d cmdliner > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-1db145.txt'' terminated with error code 2
# ocamlfind: Package `ssh' not found
# W: Field 'pkg_ssh' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d ssh > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-32cdec.txt'' terminated with error code 2
# ocamlfind: Package `maxminddb' not found
# W: Field 'pkg_maxminddb' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d maxminddb > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-60d042.txt'' terminated with error code 2
# E: Cannot find findlib package cmdliner
# E: Cannot find findlib package maxminddb
# E: Cannot find findlib package ssh
# E: Failure("3 configuration errors")
```

I let you merge the PR if you think that's fine (I haven't followed that bunch of release closely)

File: packages/tallgeese/tallgeese.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ["ocaml" "setup.ml" "-configure" "--enable-tests"]
+  ["ocaml" "setup.ml" "-build"]
+  ["ocaml" "setup.ml" "-test"]
+]
+remove: ["ocaml" "%{etc}%/tallgeese/_oasis_remove_.ml" "%{etc}%/tallgeese"]
+depends: [
+  "oasis" {build & >= "0.4"}
+  "ocamlfind" {build}
+]
+depopts: [
```
Comment (samoht) [inline packages/tallgeese/tallgeese.0.1/opam]: are these really depopts?

Comment (fxfactorial) [inline packages/tallgeese/tallgeese.0.1/opam]: Do you mean whether I really need them? Yes, they are part of the application. I guess that means they should be plain depends?

Comment (dbuenzli) [inline packages/tallgeese/tallgeese.0.1/opam]: @fxfactorial The question you should ask yourself is: is it possible to build the package without these dependencies ? If the answer is no, then these should be in `depends:`.

Comment (fxfactorial) [inline packages/tallgeese/tallgeese.0.1/opam]: @dbuenzli Okay, in that case I need to update this, should be in depends. Also thanks for all your amazing packages.

**Comment by samoht** [comment]
I've relaunched the OSX build: https://travis-ci.org/ocaml/opam-repository/jobs/84832242

**Comment by samoht** [comment]
@fxfactorial can you change the depopts into depends for this release? The OSX build fails because of that:

```
#=== ERROR while installing tallgeese.0.1 =====================================#
# opam-version 1.2.2
# os           darwin
# command      ocaml setup.ml -configure --prefix /Users/travis/.opam/system
# path         /Users/travis/.opam/system/build/tallgeese.0.1
# compiler     system (4.02.3)
# exit-code    1
# env-file     /Users/travis/.opam/system/build/tallgeese.0.1/tallgeese-2632-8b8e2c.env
# stdout-file  /Users/travis/.opam/system/build/tallgeese.0.1/tallgeese-2632-8b8e2c.out
# stderr-file  /Users/travis/.opam/system/build/tallgeese.0.1/tallgeese-2632-8b8e2c.err
### stderr ###
# [...]
# ocamlfind: Package `cmdliner' not found
# W: Field 'pkg_cmdliner' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d cmdliner > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-df76d7.txt'' terminated with error code 2
# ocamlfind: Package `ssh' not found
# W: Field 'pkg_ssh' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d ssh > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-996b74.txt'' terminated with error code 2
# ocamlfind: Package `maxminddb' not found
# W: Field 'pkg_maxminddb' is not set: Command ''/Users/travis/.opam/system/bin/ocamlfind' query -format %d maxminddb > '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/oasis-561a34.txt'' terminated with error code 2
# E: Cannot find findlib package cmdliner
# E: Cannot find findlib package maxminddb
# E: Cannot find findlib package ssh
# E: Failure("3 configuration errors")
```

**Comment by fxfactorial** [comment]
@samoht Oops, I thought I did before, now did it.

**Comment by samoht** [comment]
OSX build failed again, restarting it: https://travis-ci.org/ocaml/opam-repository/jobs/84933341

**Comment by fxfactorial** [comment]
@samoht curl/network's fault

```
+curl -OL http://xquartz.macosforge.org/downloads/SL/XQuartz-2.7.6.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:01:21 --:--:--     0
curl: (56) Recv failure: Operation timed out
```

**Comment by samoht** [comment]
Travis is so broken.... I've restarted the build again.

**Comment by fxfactorial** [comment]
@samoht Build failed, the travis machines don't have `libmaxminddb` 

```
# src/maxminddb_stubs.c:16:10: fatal error: 'maxminddb.h' file not found
# #include <maxminddb.h>
```

**Comment by samoht** [comment]
Can you add a depext to `maxminddb`? This should be something like:

```
depexts: [["osx" "homebrew"] ["libmaxminddb"]]
```

**Comment by fxfactorial** [comment]
@samoht done, have you say you have a lot of patience!

**Comment by samoht** [comment]
All green \o/

(and I'm curious about tallgeese so I want to make sure it compiles before testing it :p)

**Comment by fxfactorial** [comment]
@samoht ha! be sure to follow the gif, its a pretty rough draft but it does work.

**Pull request merged**